virttest/env_process.py

```diff
@@ -1147,11 +1147,11 @@ def postprocess(test, params, env):
 
     if kernel_extra_params_add or kernel_extra_params_remove:
         for vm in env.get_all_vms():
-            if vm:
-                if not vm.is_alive():
+            if params.get("vm_type") == "libvirt":
+                if vm and not vm.is_alive():
                     vm.start()
-                utils_test.update_boot_option(vm, args_added=kernel_extra_params_remove,
-                                              args_removed=kernel_extra_params_add)
+            utils_test.update_boot_option(vm, args_added=kernel_extra_params_remove,
+                                          args_removed=kernel_extra_params_add)
```

Comment:
I see couple of testcases in tp-qemu uses this param, so I think it is better to be supported for qemu tests as well. Also I believe it is a good to have this working in both qemu and libvirt that helps us to bringup VM and test it with different kernel commandline options.

We have placed this option before postprocessing of VMs in postprocess. So, if test uses the params and VM is dead in the context then framework should raise the error, which it does now.

Reply:
You said that raises the error is just for qemu vm, but not for libvirt vm since libvirt vm will start even if the vm is dead in the context from the above codes.
So why libvirt vm could do start vm ?  qemu vm can't.

Reply:
do you want to create VM for qemu test if vm is dead in qemu test instead of error out ?
In libvirt test, start VM will work because VM xml is defined, persistent and managed by libvirt, but in qemu test we have to handle. The only option I could think of is to create the VM and proceed incase of qemu test. Please share your thoughts